File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.cpp
// Last updated: 3/30/2026, 2:14:28 PM
class Solution {
public:
    int strStr(string haystack, string needle) {
        int hayptr = 0, needptr = 0;
        for(int index = 0 ; index < haystack.size() ; index++){
            if(haystack[index]==needle[needptr]){
                hayptr = index;
                while(haystack[hayptr]==needle[needptr] && needptr < needle.size()){
                    hayptr++;
                    needptr++;
                }
                if(needptr==needle.size()) return index;
                else{
                    needptr = 0;
                }
            }
        }
        return -1;
    }
};

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.java
// Last updated: 3/30/2026, 2:14:24 PM
class Solution {
    public int strStr(String haystack, String needle) {
        int hayptr = 0, needptr = 0;
        char[] hayarr = haystack.toCharArray();
        char[] needarr = needle.toCharArray();
        for(int index = 0 ; index < hayarr.length ; index++){
            if(hayarr[index]==needarr[needptr]){
                hayptr = index;
                while(hayptr < haystack.length() && needptr < needle.length() && hayarr[hayptr]==needarr[needptr]){
                    hayptr++;
                    needptr++;
                }
                if(needptr==needarr.length) return index;
                else needptr = 0;
            }
        }
        return -1;
    }
}